docs/dataservice-read-from-stream-layer-example.md

```diff
@@ -0,0 +1,193 @@
+# Read from a Stream Layer (Example)
+
+On this page, find instructions on how to build and run the example project on different platforms and get catalog and partition metadata, as well as partition data using the HERE Open Location Platform (OLP) SDK for C++.
+
+Before you run the example project, authorize to HERE OLP:
+
+1. On the [Apps & keys](https://platform.here.com/admin/apps) page, copy your application access key ID and access key secret.
+   For instructions on how to get the access key ID and access key secret, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.
+
+2. In `examples/main.cpp`, replace the placeholders with your access key ID, access key secret, and Here Resource Name (HRN) of the catalog.
+   **Note:**  You can also specify these values using the command line options.
+
+   ```cpp
+   AccessKey access_key{};  // Your access key ID and access key secret.
+   ```
+
+## Build and Run on Linux
+
+To build and run the example project on Linux:
+
+1. Enable examples of the `CMake` targets.
+
+   ```bash
+   mkdir build && cd build
+   cmake -DOLP_SDK_BUILD_EXAMPLES=ON ..
+   ```
+
+2. In the **build** folder, build the example project.
+
+   ```bash
+   cmake --build . --target dataservice-example
+   ```
+
+3. Execute the example project.
+
+   ```bash
+   ./examples/dataservice-example --example read --key_id "here.access.key.id" --key_secret "here.access.key.secret" --catalog "catalog"
+   ```
+
+4. (Optional) To run the example with other parameters, run the help command, and then select the needed parameter.
+
+   ```bash
+   ./examples/dataservice-example --help
+   ```
+
+After building and running the example project, you see the following information:
+
+- `edge-example-catalog` &ndash; a catalog description
+- `versioned-world-layer` &ndash; a layer description
+- `Request partition data - Success, data size - 3375` - a success message that displays the size of data retrieved from the requested partition.
```

Comment:
These are not valid for stream layer reading.
Please get some valid logs from @LiubovDidkivska.

Reply:
Discussed and changed.